### PR TITLE
✨ feat/ add and remove liquidity on uniswapV2

### PR DESCRIPTION
**Pull request by simluyt**
PR for adding and removing LP to uniswapV2 pools

### CODE REVIEW

File: src/Vault.sol [hunk 1]
```diff
@@ -780,4 +782,38 @@ contract Vault {
...
+
+    function vaultManagementAction(address _actionHandler, bytes calldata _actionData) public onlyOwner {
+        require(IMainRegistry(registryAddress).isActionAllowlisted(_actionHandler), "VL_VMA: Action is not allowlisted");
+
+        (actionAssetsData memory outgoing_, actionAssetsData memory incoming_) =
+            abi.decode(_actionData, (actionAssetsData, actionAssetsData));
+
+        // withdraw to actionHandler
+        for (uint256 i; i < outgoing_.assets.length;) {
+            outgoing_.preActionBalances[i] = IERC20(outgoing_.assets[i]).balanceOf(address(this));
```
Comment (j-vp) [inline src/Vault.sol:798]: where do we use this .preActionBalances?

File: src/AssetManagement/actions/UniswapV2LPAction.sol [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+        uint256 incomingLength = incomingAssets_.assets.length;
+        incomingAssetAmounts_ = new uint256[](incomingLength);
+        for (uint256 i; i < incomingLength;) {
+            incomingAssetAmounts_[i] =
+                IERC20(incomingAssets_.assets[i]).balanceOf(address(this)) - incomingAssets_.preActionBalances[i];
+
+            // Check incoming assets are as expected
+            require(
+                incomingAssetAmounts_[i] >= incomingAssets_.assetAmounts[i],
```
Comment (j-vp) [inline src/AssetManagement/actions/UniswapV2LPAction.sol:138]: can be removed, checked by the router

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -2,6 +2,7 @@ complete-test-run:
 
   rules:
     - if: '$CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "main"'
+    - if: '$CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "dev"'
```
Comment (j-vp) [inline .gitlab-ci.yml:5]: add this in the github workflow, not the gitlab workflow

File: src/AssetManagement/actions/UniswapV2LPAction.sol [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        /*///////////////////////////////
+                    DECODE
+        ///////////////////////////////*/
+        (_outgoing, _incoming, _selector) =
+            abi.decode(_actionSpecificData, (actionAssetsData, actionAssetsData, bytes4));
+
+        if (bytes4(_selector) == bytes4(keccak256("add"))) {
+            require(_outgoing.assets.length >= 2, "UV2A_LP: Need atleast two base tokens");
+            require(_incoming.assets.length == 1, "UV2A_LP: Can only out one lp token");
+        } else if (bytes4(_selector) == bytes4(keccak256("remove"))) {
```
[the remark this answers is not in the document]
Comment (alikonuk1) [reply src/AssetManagement/actions/UniswapV2LPAction.sol]: Added a revert here and removed the check on _execute

File: src/AssetManagement/actions/UniswapV2LPAction.sol (hunk 2 above)
Comment (alikonuk1) [reply src/AssetManagement/actions/UniswapV2LPAction.sol]: Removed.

**Comment by j-vp** [comment]
Replaced by #176

**Pull request closed without merging**